src/cr/cube/stripe/cubemeasure.py

```diff
@@ -0,0 +1,310 @@
+# encoding: utf-8
+
+"""Provides abstracted cube-measure objects used as the basis for second-order measures.
+
+There are several cube-measures that can appear in a cube-response, including
+unweighted-counts, weighted-counts (aka. counts), means, and others.
+"""
+
+from __future__ import division
+
+import numpy as np
+
+from cr.cube.enums import DIMENSION_TYPE as DT
+from cr.cube.util import lazyproperty
+
+
+class CubeMeasures(object):
+    """Provides access to all cube-measure objects for this stripe."""
+
+    def __init__(self, cube, rows_dimension, ca_as_0th, slice_idx):
+        self._cube = cube
+        self._rows_dimension = rows_dimension
+        self._ca_as_0th = ca_as_0th
+        self._slice_idx = slice_idx
+
+    @lazyproperty
+    def cube_means(self):
+        """_BaseCubeMeans subclass object for this stripe."""
+        return _BaseCubeMeans.factory(self._cube, self._rows_dimension)
+
+    @lazyproperty
+    def unweighted_cube_counts(self):
+        """_BaseUnweightedCubeCounts subclass object for this stripe."""
+        return _BaseUnweightedCubeCounts.factory(
+            self._cube, self._rows_dimension, self._ca_as_0th, self._slice_idx
+        )
+
+    @lazyproperty
+    def weighted_cube_counts(self):
+        """_BaseWeightedCubeCounts subclass object for this stripe."""
+        return _BaseWeightedCubeCounts.factory(
+            self._cube, self._rows_dimension, self._ca_as_0th, self._slice_idx
+        )
+
+
+class _BaseCubeMeasure(object):
+    """Base class for all cube-measure objects."""
+
+    def __init__(self, rows_dimension):
+        self._rows_dimension = rows_dimension
+
+
+# === MEANS ===
+
+
+class _BaseCubeMeans(_BaseCubeMeasure):
+    """Base class for means cube-measure variants."""
+
+    def __init__(self, rows_dimension, means):
+        super(_BaseCubeMeans, self).__init__(rows_dimension)
+        self._means = means
+
+    @classmethod
+    def factory(cls, cube, rows_dimension):
+        """Return _BaseCubeMeans subclass instance appropriate to `cube`."""
+        # --- TODO: note that `cube.counts` is improperly overloaded to return means
+        # --- when cube.has_means. This needs to be fixed.
+        means = cube.counts
+        MeansCls = (
+            _MrCubeMeans if rows_dimension.dimension_type == DT.MR else _CatCubeMeans
+        )
+        return MeansCls(rows_dimension, means)
+
+    @lazyproperty
+    def means(self):
+        """1D np.float64 ndarray of mean for each stripe row."""
+        raise NotImplementedError("`%s` must implement `.means`" % type(self).__name__)
```

Comment:
no cover also here

Reply:
These are all patched up except for three skip and xfail tests that need the numeric-array changes in your branch. Be sure to check those and get them back working again when you get rebased onto this one.